### DISCUSSION TITLE
Can not strart Frontned and Backend e.g. APIServer\api

### DISCUSSION THREAD

**Discussion by goodjobgamie** [damikun/trouble-training#17] [Q&A]
i can not strart Frontned and Backend. I have do everything follow this link https://github.com/damikun/trouble-training/blob/main/Doc/Manual_Install.md
my everonment : Windows 10 , wsl2, docker, 
dotnet --list-sdks
2.2.300 [C:\Program Files\dotnet\sdk]
5.0.302 [C:\Program Files\dotnet\sdk]
6.0.301 [C:\Program Files\dotnet\sdk]

when i try to start app , i got this error

D:\Works\trouble-training-main\src\APIServer\api> dotnet run

D:\Works\trouble-training-main\src\APIServer\Aplication\Common\Behaviours\PerformanceBehaviour.cs(67,48): warning CS069
3: Type parameter 'TRequest' has the same name as the type parameter from outer type 'PerformanceBehaviour<TRequest, TR
esponse>' [D:\Works\trouble-training-main\src\APIServer\Aplication\APIServer.Aplication.csproj]
D:\Works\trouble-training-main\src\APIServer\Aplication\Graphql\Extensions\UseConnection\UseConnection.cs(76,39): warni
ng CS8600: Converting null literal or possible null value to non-nullable type. [D:\Works\trouble-training-main\src\API
Server\Aplication\APIServer.Aplication.csproj]
[  User profile is available. Using 'C:\Users\kengh\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
[  Sensitive data logging is enabled. Log entries and exception messages may include sensitive application data; this mode should only be enabled during development.
[  The property 'WebHook.HookEvents' is a collection or enumeration type with a value converter but with no value comparer. Set a value comparer to ensure the collection/enumeration elements are compared correctly.
[  Entity Framework Core 6.0.0 initialized 'ApiDbContext' using provider 'Microsoft.EntityFrameworkCore.Sqlite:6.0.0' with options: SensitiveDataLoggingEnabled DetailedErrorsEnabled MaxPoolSize=1024
[  Executed DbCommand (22ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
SELECT COUNT(*) FROM "sqlite_master" WHERE "type" = 'table' AND "rootpage" IS NOT NULL;
[  Starting Hangfire Server using job storage: 'Connection string: Hangfire.db,  prefix: hangfire'
[  Using the following options for Hangfire Server:
    Worker count: 20
    Listening queues: 'default'
    Shutdown timeout: 00:00:15
    Schedule polling interval: 00:00:15
[  Server kenghot-hp:14000:69089965 successfully announced in 148.0882 ms
[  Server kenghot-hp:14000:69089965 is starting the registered dispatchers: ServerWatchdog, ServerJobCancellationWatcher, ExpirationManager, CountersAggregator, Worker, DelayedJobScheduler, RecurringJobScheduler...
[  Server kenghot-hp:14000:69089965 all the dispatchers started
[  Failed to process the job '694': an exception occurred. Retry attempt 4 of 5 will be performed in 00:02:32.
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'IServiceProvider'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ThrowHelper.ThrowObjectDisposedException()
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateScope()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.CreateScope()
   at Hangfire.AspNetCore.AspNetCoreJobActivator.BeginScope(JobActivatorContext context)
   at Hangfire.JobActivator.BeginScope(PerformContext context)
   at Hangfire.Server.CoreBackgroundJobPerformer.Perform(PerformContext context)
   at Hangfire.Server.BackgroundJobPerformer.<>c__DisplayClass9_0.<PerformJobWithFilters>b__0()
   at Hangfire.Server.BackgroundJobPerformer.InvokePerformFilter(IServerFilter filter, PerformingContext preContext, Func`1 continuation)
   at Hangfire.Server.BackgroundJobPerformer.<>c__DisplayClass9_1.<PerformJobWithFilters>b__2()
   at Hangfire.Server.BackgroundJobPerformer.PerformJobWithFilters(PerformContext context, IEnumerable`1 filters)
   at Hangfire.Server.BackgroundJobPerformer.Perform(PerformContext context)
   at Hangfire.Server.Worker.PerformJob(BackgroundProcessContext context, IStorageConnection connection, String jobId)
[  Failed to process the job '693': an exception occurred. Retry attempt 4 of 5 will be performed in 00:01:48.
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'IServiceProvider'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ThrowHelper.ThrowObjectDisposedException()
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateScope()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.CreateScope()
   at Hangfire.AspNetCore.AspNetCoreJobActivator.BeginScope(JobActivatorContext context)
   at Hangfire.JobActivator.BeginScope(PerformContext context)
   at Hangfire.Server.CoreBackgroundJobPerformer.Perform(PerformContext context)
   at Hangfire.Server.BackgroundJobPerformer.<>c__DisplayClass9_0.<PerformJobWithFilters>b__0()
   at Hangfire.Server.BackgroundJobPerformer.InvokePerformFilter(IServerFilter filter, PerformingContext preContext, Func`1 continuation)
   at Hangfire.Server.BackgroundJobPerformer.<>c__DisplayClass9_1.<PerformJobWithFilters>b__2()
   at Hangfire.Server.BackgroundJobPerformer.PerformJobWithFilters(PerformContext context, IEnumerable`1 filters)
   at Hangfire.Server.BackgroundJobPerformer.Perform(PerformContext context)
   at Hangfire.Server.Worker.PerformJob(BackgroundProcessContext context, IStorageConnection connection, String jobId)
[  Runtime unhandled exception
Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException: The system cannot find the path specified.
   at Internal.Cryptography.Pal.CertificatePal.FromBlobOrFile(ReadOnlySpan`1 rawData, String fileName, SafePasswordHandle password, X509KeyStorageFlags keyStorageFlags)
   at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(String fileName, String password, X509KeyStorageFlags keyStorageFlags)
   at System.Security.Cryptography.X509Certificates.X509Certificate2..ctor(String fileName, String password)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Certificates.CertificateConfigLoader.LoadCertificate(CertificateConfig certInfo, String endpointName)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelConfigurationLoader.LoadDefaultCert()
   at Microsoft.AspNetCore.Server.Kestrel.KestrelConfigurationLoader.Reload()
   at Microsoft.AspNetCore.Server.Kestrel.KestrelConfigurationLoader.Load()
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at APIServer.Program.Main(String[] args) in D:\Works\trouble-training-main\src\APIServer\api\Program.cs:line 22
[  Stopping the server due to DomainUnload or ProcessExit event...
[  Server kenghot-hp:14000:69089965 caught stopping signal...
[  Server kenghot-hp:14000:69089965 caught stopped signal...
[  Server kenghot-hp:14000:69089965 caught shutdown signal...
[  Server kenghot-hp:14000:69089965 All dispatchers stopped
[  Server kenghot-hp:14000:69089965 successfully reported itself as stopped in 3.8036 ms
[  Server kenghot-hp:14000:69089965 has been stopped in total 8.849 ms

**Comment by damikun** [comment]
It will not run in docker since WSL blocks some behaviour such as certificate instalation and adding paths.. 

Docker also isolates containers to use host.docker.internall....  The APIServer, IdentityServer and BFF must be run manualy or using Tye... Since they require working HTTPS...

Try to build directly on Windows using install scripts... (Install script = Nuke). The manual install was not used on workshop i dont know the state...., insted automatic script was used..

Follow steps on main readme....

-> Use prepared init script. Run `.\build.cmd E2E_RunAs_Local` from project root. (This will take time since restore)
-> From root directory run `tye run`

**Reply by damikun** [reply]
If you look on CI/CD you can see the github can build and run the project...

**Reply by damikun** [reply]
You can ignore errors on startup with messages _"Failed to process the job '***': an exception occurred."_ thats from scheduler, informing that job failed and will be re-queued this happends if app is restarted during startup process.... Its just information log...

**Reply by damikun** [reply]
The main issue is ```X509KeyStorageFlags keyStorageFlags) at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(String fileName, String password, X509KeyStorageFlags keyStorageFlags) at System.Security.Cryptography.X509Certificates.X509Certificate2..ctor(String fileName, String password) at``` which says you cannot comunicate between services using HTTPS... Since the paths or certificates are not properly installed...

**Reply by goodjobgamie** [reply]
I've got another error after .\build.cmd E2E_RunAs_Local and then 
 tye run

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! trouble-training-frontend@0.1.0 build: `craco build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the trouble-training-frontend@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\kengh\AppData\Roaming\npm-cache\_logs\2022-07-21T11_28_25_645Z-debug.log
ProcessException: Process 'npm.cmd' exited with code 1.
   > "C:\Program Files\nodejs\npm.cmd" run build
   @ D:\Works\trouble-training-main\Src\BFF\API\ClientApp
Error output:
   npm ERR! code ELIFECYCLE
   npm ERR! errno 1
   npm ERR! trouble-training-frontend@0.1.0 build: `craco build`
   npm ERR! Exit status 1
   npm ERR!
   npm ERR! Failed at the trouble-training-frontend@0.1.0 build script.
   npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

   npm ERR! A complete log of this run can be found in:
   npm ERR!     C:\Users\kengh\AppData\Roaming\npm-cache\_logs\2022-07-21T11_28_25_645Z-debug.log


and debug.log  file is:
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build'
1 verbose cli ]
2 info using npm@6.14.14
3 info using node@v14.17.5
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle trouble-training-frontend@0.1.0~prebuild: trouble-training-frontend@0.1.0
6 info lifecycle trouble-training-frontend@0.1.0~build: trouble-training-frontend@0.1.0
7 verbose lifecycle trouble-training-frontend@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle trouble-training-frontend@0.1.0~build: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;D:\Works\trouble-training-main\Src\BFF\API\ClientApp\node_modules\.bin;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn\;C:\Program Files\Azure Data Studio\bin;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Program Files (x86)\Bitvise SSH Client;D:\Tools\xampp\php;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Program Files (x86)\dotnet\;C:\Users\kengh\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\kengh\AppData\Local\Programs\Python\Python39\;C:\Users\kengh\AppData\Local\Microsoft\WindowsApps;C:\Users\kengh\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\kengh\.dotnet\tools;C:\Users\kengh\AppData\Roaming\npm;C:\Users\kengh\AppData\Local\GitHubDesktop\bin;C:\Program Files\heroku\bin;C:\Program Files\wkhtmltopdf\bin;C:\Program Files\PostgreSQL\13\bin;d:\Tools\flutter\bin;d:\tools\php;C:\Users\kengh\AppData\Local\Pub\Cache\bin;c:\dapr
9 verbose lifecycle trouble-training-frontend@0.1.0~build: CWD: D:\Works\trouble-training-main\Src\BFF\API\ClientApp
10 silly lifecycle trouble-training-frontend@0.1.0~build: Args: [ '/d /s /c', 'craco build' ]
11 silly lifecycle trouble-training-frontend@0.1.0~build: Returned: code: 1  signal: null
12 info lifecycle trouble-training-frontend@0.1.0~build: Failed to exec build script
13 verbose stack Error: trouble-training-frontend@0.1.0 build: `craco build`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:400:28)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:400:28)
13 verbose stack     at maybeClose (internal/child_process.js:1055:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
14 verbose pkgid trouble-training-frontend@0.1.0
15 verbose cwd D:\Works\trouble-training-main\Src\BFF\API\ClientApp
16 verbose Windows_NT 10.0.19044
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
18 verbose node v14.17.5
19 verbose npm  v6.14.14
20 error code ELIFECYCLE
21 error errno 1
22 error trouble-training-frontend@0.1.0 build: `craco build`
22 error Exit status 1
23 error Failed at the trouble-training-frontend@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

Target                        Status      Duration
──────────────────────────────────────────────────
Identity_Clean                Succeeded     < 1sec
Device_Clean                  Succeeded       0:17
Frontend_Clean                Succeeded       0:15
BFF_Clean                     Succeeded     < 1sec
API_Clean                     Succeeded     < 1sec
Clean                         Succeeded     < 1sec
Identity_Restore              Succeeded       0:05
Identity_Compile              Succeeded       0:14
Frontend_Restore              Succeeded       2:03
Device_Restore_Frontend       Succeeded       0:46
Device_Restore                Succeeded       0:02
Device_Compile                Succeeded       0:03
Restore_Tools                 Succeeded     < 1sec
Frontend_RelayCompile         Succeeded       0:02
Frontend_AddTailwind          Succeeded       0:15
Frontend_TryBuild             Failed          0:46   // ProcessException: Process 'npm.cmd' exited with code 1.
Device_ALL                    NotRun              
Frontend_All                  NotRun              
Identity_All                  NotRun              
BFF_Restore                   NotRun              
BFF_Compile                   NotRun              
BFF_All                       NotRun              
API_Restore                   NotRun              
API_Compile                   NotRun              
API_APITest                   NotRun              
API_IntegrationTest           NotRun              
API_UnitTest                  NotRun              
API_Test                      NotRun              
API_All                       NotRun              
All                           NotRun              
Identity_Migrate_DB           NotRun              
Identity_Init                 NotRun              
API_Migrate_DB                NotRun              
API_Init                      NotRun              
Init                          NotRun              
Clean_Dev_Certificates        NotRun              
SetupDevCertificates          NotRun              
TrustDevCertificates_Local    NotRun              
SetupCertificates_Local       NotRun              
Start_Identity_Server         NotRun              
Init_Newman                   NotRun              
Start_BFF_Server              NotRun              
Start_API_Server              NotRun              
E2E_RunAs_Local               NotRun              
Stop_Identity_Server          NotRun              
Stop_BFF_Server               NotRun              
Stop_API_Server               NotRun

**Reply by damikun** [reply]
What version of node you have? TLS or Latest?  Im using TLS the latest usualy has some inconsistency for package install/restore...

**Reply by damikun** [reply]
I will have a look on evening maybe some package gets updated in time and there is broken dep... Will ping you hire and i will retriger now the CI/CD to make sure...

**Reply by goodjobgamie** [reply]
> What version of node you have? TLS or Latest? Im using TLS the latest usualy has some inconsistency for package install/restore...

v14.17.5

**Reply by goodjobgamie** [reply]
I forked you code and try to CI/CD then got this error :
Frontend_AddTailwind
Repeating warnings and errors:
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated har-validator@5.1.3: this library is no longer supported
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated @opentelemetry/exporter-collector@0.25.0: Please use trace and metric specific exporters @opentelemetry/exporter-trace-otlp-http and @opentelemetry/exporter-metrics-otlp-http
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @cypress/react@5.10.3
npm ERR! Found: react@18.0.0-rc.0-next-9d4e8e84f-20220207
npm ERR! node_modules/react
npm ERR!   peer react@"*" from @testing-library/react@11.1.0
npm ERR!   node_modules/@testing-library/react
npm ERR!     @testing-library/react@"11.1.0" from the root project
npm ERR!   peer react@"18.0.0-rc.0-next-9d4e8e84f-20220207" from react-dom@18.0.0-rc.0-next-9d4e8e84f-20220207
npm ERR!   node_modules/react-dom
npm ERR!     peer react-dom@"*" from @testing-library/react@11.1.0
npm ERR!     node_modules/@testing-library/react
npm ERR!       @testing-library/react@"11.1.0" from the root project
npm ERR!     react-dom@"18.0.0-rc.0-next-9d4e8e84f-20220207" from the root project
npm ERR!   1 more (the root project)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^=16.x || ^=17.x" from @cypress/react@5.10.3
npm ERR! node_modules/@cypress/react
npm ERR!   dev @cypress/react@"5.10.3" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@"^=16.x || ^=17.x" from @cypress/react@5.10.3
npm ERR!   node_modules/@cypress/react
npm ERR!     dev @cypress/react@"5.10.3" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See C:\npm\cache\eresolve-report.txt for a full report.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\2022-07-21T11_28_28_945Z-debug-0.log
ProcessException: Process 'npm.cmd' exited with code 1.
   > "C:\Program Files\nodejs\npm.cmd" install -D tailwindcss@npm:@tailwindcss/postcss7-compat@2.2.17 postcss@7.0.39 autoprefixer@9.8.8
   @ D:\a\trouble-training\trouble-training\Src\BFF\API\ClientApp
Error output:
   npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
   npm ERR! code ERESOLVE
   npm ERR! ERESOLVE could not resolve
   npm ERR! 
   npm ERR! While resolving: @cypress/react@5.10.3
   npm ERR! Found: react@18.0.0-rc.0-next-9d4e8e84f-20220207
   npm ERR! node_modules/react
   npm ERR!   peer react@"*" from @testing-library/react@11.1.0
   npm ERR!   node_modules/@testing-library/react
   npm ERR!     @testing-library/react@"11.1.0" from the root project
   npm ERR!   peer react@"18.0.0-rc.0-next-9d4e8e84f-20220207" from react-dom@18.0.0-rc.0-next-9d4e8e84f-20220207
   npm ERR!   node_modules/react-dom
   npm ERR!     peer react-dom@"*" from @testing-library/react@11.1.0
   npm ERR!     node_modules/@testing-library/react
   npm ERR!       @testing-library/react@"11.1.0" from the root project
   npm ERR!     react-dom@"18.0.0-rc.0-next-9d4e8e84f-20220207" from the root project
   npm ERR!   1 more (the root project)
   npm ERR! 
   npm ERR! Could not resolve dependency:
   npm ERR! peer react@"^=16.x || ^=17.x" from @cypress/react@5.10.3
   npm ERR! node_modules/@cypress/react
   npm ERR!   dev @cypress/react@"5.10.3" from the root project
   npm ERR! 
   npm ERR! Conflicting peer dependency: react@17.0.2
   npm ERR! node_modules/react
   npm ERR!   peer react@"^=16.x || ^=17.x" from @cypress/react@5.10.3
   npm ERR!   node_modules/@cypress/react
   npm ERR!     dev @cypress/react@"5.10.3" from the root project
   npm ERR! 
   npm ERR! Fix the upstream dependency conflict, or retry
   npm ERR! this command with --force, or --legacy-peer-deps
   npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
   npm ERR! 
   npm ERR! See C:\npm\cache\eresolve-report.txt for a full report.
   npm ERR! A complete log of this run can be found in:
   npm ERR!     C:\npm\cache\_logs\2022-07-21T11_28_28_945Z-debug-0.log
